File: bidrag-commons-test/src/main/kotlin/no/nav/bidrag/commons/testdatagenerering/navn/TestNavn.java
package no.nav.bidrag.commons.testdatagenerering.navn;

import lombok.Data;
import no.nav.bidrag.commons.testdatagenerering.PeriodisertTestData;

import java.time.LocalDate;

@Data
public class TestNavn implements PeriodisertTestData {
    private final String fornavn;
    private final String etternavn;
    private final LocalDate periodeFra;
    private final LocalDate periodeTil;

    public String getSammensatt() {
        return getEtternavn() + ", " + getFornavn();
    }
}

File: bidrag-generer-testdata/src/main/kotlin/no.nav.bidrag.generer.testdata/navn/TestNavn.kt
package no.nav.bidrag.generer.testdata.navn

import no.nav.bidrag.generer.testdata.PeriodisertTestData
import java.time.LocalDate

@Suppress("unused")
data class TestNavn(
    val fornavn: String? = null,
    val etternavn: String? = null,
    override val periodeFra: LocalDate? = null,
    override val periodeTil: LocalDate? = null,
) : PeriodisertTestData {
    val sammensatt: String
        get() = "$etternavn, $fornavn"
}
